packages/itwinui-react/src/react-table/react-table-types.test.tsx

```diff
@@ -0,0 +1,802 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+/**
+ * This test file to test some Table type related cases that have not been tested in other files
+ * (e.g. storybook, unit tests, etc.)
+ */
+
+// To hide impertinent errors
+/* eslint-disable @typescript-eslint/no-unused-vars */
+/* eslint-disable @typescript-eslint/ban-ts-comment */
+
+import {
+  ActionColumn,
+  Table,
+  TablePaginator,
+  tableFilters,
+  EditableCell,
+  DefaultCell,
+  SelectionColumn,
+  ExpanderColumn,
+} from '../core/Table/index.js';
+import type {
+  TableFilterValue,
+  TablePaginatorRendererProps,
+} from '../core/Table/index.js';
+
+// Table types can be imported with an alias or selective types can be imported too.
+import * as _TableTypes from 'src/react-table/react-table.js';
+import type { Column } from 'src/react-table/react-table.js';
+
+// Can import types from file or by package name.
+import * as TableTypes from 'react-table';
+
+type TableProps<T extends Record<string, unknown> = Record<string, unknown>> =
+  React.ComponentProps<typeof Table<T>>;
+
+import React from 'react';
+import {
+  Anchor,
+  DropdownMenu,
+  IconButton,
+  Input,
+  MenuItem,
+  Text,
+  Tooltip,
+} from 'src/core/index.js';
+import { SvgMore } from 'src/core/utils/index.js';
+
+/**
+ * Confirm that `satisfies` on columns does not give any unnecessary type errors.
+ */
+() => {
+  const columns = [
+    {
+      Header: 'Header 1',
+      accessor: 'header1',
+    },
+    {
+      Header: 'Header 2',
+      accessor: 'header2',
+    },
+  ] satisfies TableTypes.Column<{
+    header1: string;
+    header2: string;
+  }>[];
+
+  const data = [
+    {
+      header1: 'row1',
+      header2: 'row1',
+    },
+    {
+      header1: 'row2',
+      header2: 'row2',
+    },
+  ];
+
+  return <Table columns={columns} data={data} emptyTableContent='No data.' />;
+};
+
+/**
+ * To make sure we cannot pass nested columns.
+ */
+() => {
+  // Don't show TS warning: 'columns' is declared but its value is never read.ts(6133)
+  // @ts-ignore - Ignore declared but unused variable.
+  const columns = [
+    {
+      Header: 'Header 1',
+      accessor: 'header1',
+      // There should be error: TS 2353
+      // @ts-expect-error (TS 2353): `columns` should not exist inside of a column object.
+      columns: [
+        {
+          Header: 'Header 2',
+          accessor: 'header2',
+        },
+      ],
+    },
+  ] satisfies TableTypes.Column<{
+    header1: string;
+    header2: string;
+  }>[];
+};
+
+/**
+ * Confirm that `Column` and `TableTypes.Column` are equivalent.
+ */
+() => {
```

Comment:
i don't think this one is testing anything. `TableTypes.Column` and `Column` are quite literally the same type imported from the same file

Reply:
I meant to test if the types from `react-table` and `src/react-table/react-table.js` are equivalent. This was to confirm whether the `paths` section of the tsconfig was working correctly.

![image](https://github.com/iTwin/iTwinUI/assets/45748283/84549028-f7e9-4476-b4b0-0253300a06ab)

Yes, the comment was misleading, so I improved the comment (https://github.com/iTwin/iTwinUI/pull/1514/commits/2cd4729b017fe1ef7c55ac523a7303fb6436672d).

Reply:
but it's still not testing anything. the path alias is a development-only detail and disappears in the published output, making this test pointless

Reply:
Yes, it is a development only detail. `tsc --noEmit` will check for this development only detail.

E.g. I was hoping for an error like this if the `paths` in tsconfig was somehow pointed to a wrong file and thus had some incorrect types.

![image](https://github.com/iTwin/iTwinUI/assets/45748283/2cc8e649-64f8-47cf-96c1-1c84961822e1)
![image](https://github.com/iTwin/iTwinUI/assets/45748283/cc276af8-4b4d-4e47-b428-f783ddd41078)

Reply:
if it was pointed to a wrong file, then the types inside our Table source code would not work and the build should fail

Reply:
Makes sense, removed (https://github.com/iTwin/iTwinUI/pull/1514/commits/94c3e54d2cc0601168a938d89e738c70ba2bcd29).